remote-processor/RemoteCommandHandlerTemplate.h

```diff
@@ -112,144 +137,97 @@ class TRemoteCommandHandlerTemplate : public IRemoteCommandHandler
         }
 
     private:
-        std::string _strCommandName;
-        RemoteCommandParser _pfnParser;
-        uint32_t _uiMinArgumentCount;
-        std::string _strHelp;
-        std::string _strDescription;
-    };
-
-public:
-    TRemoteCommandHandlerTemplate(CCommandParser* pCommandParser) : _pCommandParser(pCommandParser), _uiMaxCommandUsageLength(0)
-    {
-        // Help Command
-        addCommandParser("help", NULL, 0, "", "Show commands description and usage");
-    }
-    ~TRemoteCommandHandlerTemplate()
-    {
-        uint32_t uiIndex;
 
-        for (uiIndex = 0; uiIndex < _remoteCommandParserVector.size(); uiIndex++) {
+        /** Command Handler callback */
+        RemoteCommandParser mParser;
 
-            delete _remoteCommandParserVector[uiIndex];
-        }
-    }
+        /** Needed arguments number */
+        uint32_t mMinArgumentCount;
 
-    // Parsers
-    bool addCommandParser(const std::string& strCommandName,
-        RemoteCommandParser pfnParser,
-        uint32_t uiMinArgumentCount,
-        const std::string& strHelp,
-        const std::string& strDescription)
-    {
-        if (findCommandParserItem(strCommandName)) {
+        /** Command help string */
+        std::string mHelp;
 
-            // Already exists
-            return false;
-        }
+        /** Command description string */
+        std::string mDescription;
+    };
 
-        // Add command
-        _remoteCommandParserVector.push_back(new CRemoteCommandParserItem(strCommandName, pfnParser, uiMinArgumentCount, strHelp, strDescription));
+    /** Remote command parser container type */
+    typedef std::map<std::string, RemoteCommandParserItem> RemoteCommandParserItems;
 
-        return true;
+    /**
+     * @param commandParser command parser used for command handling
+     * @param remoteCommandParserItems supported command parser items
+     */
+    RemoteCommandHandlerTemplate(CommandParser& commandParser,
+                                 RemoteCommandParserItems& remoteCommandParserItems) :
+        _commandParser(commandParser), _remoteCommandParserItems(remoteCommandParserItems)
+    {
+        /* Add the help command and specialize the help function
+         * to match RemoteCommandParser prototype
+         */
+        using namespace std::placeholders;
+        _remoteCommandParserItems.emplace(
+                "help",
+                RemoteCommandParserItem(
+                    std::bind(&RemoteCommandHandlerTemplate<CommandParser>::help, this, _1, _2, _3),
+                    0, "", "Show commands description and usage"));
```

Comment:
`[this](CommandParser&, const IRemoteCommand&, std::string& result) -> { help(result); }`
help will no longer need 2 first useless parameters!